core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -141,4 +136,41 @@ private static String stripTrailingSlash(String path) {
     }
     return result;
   }
+
+  /**
+   * Get the data file location. For the {@link DefaultLocationProvider}, the priority level are
+   * "write.data.path" -> "write.folder-storage.path" -> "table-location/data".
+   * For the {@link ObjectStoreLocationProvider}, the priority level are
+   * "write.data.path" -> "write.object-storage.path" -> "write.folder-storage.path" -> "table-location/data".
+   */
+  private static String dataLocation(Map<String, String> properties, String tableLocation, boolean isObjectStore) {
```

Comment:
Nit / non-blocking: Naming this `isObjectStorageLocationProvider` or something to indicate that it's not just object storage, but specifically the usage of object storage location provider.

However, since I just realized this is already in `LocationProviders.java`, I think this is probably fine and no need to change (especially as it's documented in the java doc).